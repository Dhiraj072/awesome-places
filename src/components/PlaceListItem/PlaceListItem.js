import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';

export default (props) => (
    <TouchableOpacity onPress={props.onPlaceListItemPress}>
        <View style={styles.listItem}>
            <Text>{props.name}</Text>
        </View>
    </TouchableOpacity>
);

const styles = StyleSheet.create({
    listItem: {
        width: '100%',
        padding: 10,
        marginBottom: 5,
        backgroundColor: '#eee',
    },
});
