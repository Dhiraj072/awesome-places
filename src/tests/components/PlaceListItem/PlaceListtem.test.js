import React from 'react';
import { shallow } from 'enzyme';
import PlaceListItem from '../../../components/PlaceListItem/PlaceListItem';

describe('PlaceListItem', () => {
    const onPlaceListItemPress = jest.fn();
    const wrapper = shallow(<PlaceListItem
        onPlaceListItemPress={onPlaceListItemPress}
    />);
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onPlaceListItemPress prop when pressed', () => {
        wrapper.find('TouchableOpacity').simulate('press');
        expect(onPlaceListItemPress).toHaveBeenCalled();
    });
});

