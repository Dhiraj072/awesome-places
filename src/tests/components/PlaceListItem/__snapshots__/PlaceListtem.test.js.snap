// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceListItem should render correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#eee",
        "marginBottom": 5,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</TouchableOpacity>
`;
